### PR TITLE
Create initials models.py based on ERD

### PR DESCRIPTION
**Pull request by e1mo**
There are a few TODOs left, or things where I'd like to discuss the naming. I'll try to add those as comments to the files tomorrow.

### CODE REVIEW

File: src/chaosinventory/base/models.py [hunk 1]
```diff
@@ -1,3 +1,198 @@
...
+    tags = models.ManyToManyField(
+        'Tag'
+    )
+
+    inventory_id = models.ForeignKey(
+        'InventoryId',
+        on_delete=models.RESTRICT,
+    )
+
+class InventoryIdSchema(CommonModel):
```
Comment (e1mo) [inline src/chaosinventory/base/models.py:155]: How do we identify different InventoryIdSchemas?

I would propose another field `name`, referring to a plugins module name and possibly sub-classes.

E.g. `chaosinventory.base.schemes.counting`, `chaosinventory.base.schemes.letters`, ...

Comment (e1mo) [reply src/chaosinventory/base/models.py:155]: Okay my bad, it inherits `name` and `note` from CommonModel...

File: src/chaosinventory/base/models.py [hunk 2]
```diff
@@ -1,3 +1,198 @@
...
+        null = True,
+        blank = True,
+    )
+
+    product = models.ForeignKey(
+        'Product',
+        on_delete=models.RESTRICT
+    )
+
+    in_item = models.ForeignKey(
```
[the remark this answers is not in the document]
Comment (clerie) [reply src/chaosinventory/base/models.py:130]: Maybe `should_be_in_item`

File: src/chaosinventory/base/models.py [hunk 3]
```diff
@@ -1,3 +1,198 @@
...
+        'Item',
+        on_delete=models.RESTRICT,
+        related_name='contents',
+        null = True,
+        blank = True,
+    )
+
+    current_item = models.ForeignKey(
```
[the remark this answers is not in the document]
Comment (clerie) [reply src/chaosinventory/base/models.py:138]: Maybe `is_currently_in_item`

File: src/chaosinventory/base/models.py [hunk 4]
```diff
@@ -1,3 +1,202 @@
...
+        related_name='current_contents',
+        null = True,
+        blank = True,
+    )
+
+    tags = models.ManyToManyField(
+        'Tag'
+    )
+
+    inventory_id = models.ForeignKey(
```
Comment (clerie) [inline src/chaosinventory/base/models.py:150]: Items should get more than one inventory id

File: src/chaosinventory/base/models.py [hunk 5]
```diff
@@ -1,3 +1,202 @@
...
+        'Overlay',
+        on_delete = models.RESTRICT,
+    )
+
+    item = models.ForeignKey(
+        'Item',
+        on_delete = models.CASCADE,
+    )
+
+    in_item = models.ForeignKey(
```
Comment (clerie) [inline src/chaosinventory/base/models.py:189]: `s/in_item/target_item/`

File: src/chaosinventory/base/models.py [hunk 6]
```diff
@@ -1,3 +1,202 @@
...
+        'Item',
+        on_delete = models.RESTRICT,
+        related_name = 'overlay_contents',
+        null = True,
+        blank = True,
+    )
+
+    location = models.ForeignKey(
```
Comment (clerie) [inline src/chaosinventory/base/models.py:197]: `s/location/target_location/`

File: src/chaosinventory/base/models.py [hunk 7]
```diff
@@ -1,3 +1,215 @@
...
+        related_name='actual_contents',
+        null=True,
+        blank=True,
+    )
+
+    tags = models.ManyToManyField(
+        'Tag'
+    )
+
+    inventory_ids = models.ManyToManyField(
```
Comment (clerie) [inline src/chaosinventory/base/models.py:160]: This field should be a `ForeignKey` too.

File: src/chaosinventory/base/models.py [hunk 8]
```diff
@@ -1,3 +1,215 @@
...
+
+    target_item = models.ForeignKey(
+        'Item',
+        on_delete=models.RESTRICT,
+        related_name='overlay_contents',
+        null=True,
+        blank=True,
+    )
+
+    target_location = models.ForeignKey(
```
Comment (clerie) [inline src/chaosinventory/base/models.py:210]: `target_location` should get a `related_name` definition too.

File: src/chaosinventory/base/models.py [hunk 9]
```diff
@@ -1,3 +1,215 @@
 from django.db import models
 
-# Create your models here.
+
+class CommonModel(models.Model):
+    name = models.CharField(max_length=255)
```
Comment (clerie) [inline src/chaosinventory/base/models.py:5]: Line break is not consistent.

Comment (clerie) [reply src/chaosinventory/base/models.py:5]: Put `max_length` in a new line.

**Review by clerie — CHANGES_REQUESTED** [review]
There are some minor errors, but overall it looks good.

File: src/chaosinventory/base/models.py [hunk 10]
```diff
@@ -1,3 +1,215 @@
...
+    type = models.ForeignKey(
+        'DataType',
+        on_delete=models.RESTRICT
+    )
+
+    class Meta:
+        abstract = True
+
+
+class Tag(models.Model):
```
[the remark this answers is not in the document]
Comment (e1mo) [reply src/chaosinventory/base/models.py:24]: _Discussed in our last conference, Tag does not require a note_

**Review by clerie — APPROVED** [review]

**Review by adnidor — APPROVED** [review]

**Review by em0lar — APPROVED** [review]

**Pull request merged**